apps/stable_diffusion/web/ui/stablelm_ui.py

```diff
@@ -169,10 +171,53 @@ def chat(
         # get iree flags that need to be overridden, from commandline args
         _extra_args = []
         # vulkan target triple
-        if args.iree_vulkan_target_triple != "":
+        vulkan_target_triple = args.iree_vulkan_target_triple
+        from shark.iree_utils.vulkan_utils import (
+            get_all_vulkan_devices,
+            get_vulkan_target_triple,
+        )
+
+        if device == "vulkan" and vulkan_target_triple == "":
+            # We already have the device_id extracted via WebUI, so we directly use
+            # that to find the target triple.
+            vulkaninfo_list = get_all_vulkan_devices()
+            vulkan_target_triple = get_vulkan_target_triple(
+                vulkaninfo_list[device_id]
+            )
+            _extra_args.append(
+                f"-iree-vulkan-target-triple={vulkan_target_triple}"
+            )
+            if "rdna" in vulkan_target_triple:
+                flags_to_add = [
+                    "--iree-spirv-index-bits=64",
+                ]
+                _extra_args = _extra_args + flags_to_add
+        elif device == "vulkan" and vulkan_target_triple != "":
```

Comment:
nit: Condition on `device == "vulkan"` once and fetch the vulkan target triple if not already set. Then you should only have to specify the extra flags to add once.

Reply:
I've made the appropriate change - you may re-review because the `push` of the update dismissed your review. :-P

Reply:
I feel like we should change that setting on the repo. Not sure that having to restamp over and over again is helpful :/